abdm/tasks/retry_failed_care_contexts.py

```diff
@@ -32,11 +35,35 @@ def retry_failed_care_contexts():
         if not patient:
             continue
 
+        patients_transactions = filtered_transactions.filter(
+            meta_data__hf_id=transaction_query["hf_id"],
+            meta_data__abha_number=transaction_query["abha_number"],
+        )
+
+        care_contexts = []
+        for transaction in patients_transactions:
+            if transaction.meta_data.get("type") != "hip_initiated_linking":
+                continue
+
+            for care_context_reference in transaction.meta_data.get("care_contexts"):
+                care_context = care_context_dict_from_reference_id(
+                    care_context_reference
+                )
+
+                if care_context:
+                    care_contexts.append(care_context)
+
+            transaction.status = TransactionStatus.CANCELLED
```

Comment:
These are being marked as cancelled, and will get filtered in case the below api call fails